api/v1alpha1/managedcluster_types.go

```diff
@@ -98,10 +78,24 @@ type ManagedClusterSpec struct {
 	// DryRun specifies whether the template should be applied after validation or only validated.
 	DryRun     bool   `json:"dryRun,omitempty"`
 	Credential string `json:"credential,omitempty"`
+	// Tier sets the priority for the services defined in this spec.
+	// Higher value means lower priority and vice versa.
+	// In case of conflict with another object managing the service,
+	// the one with lower priority will get to deploy its services.
+	// +kubebuilder:default:=100
+	// +kubebuilder:validation:Minimum=1
+	// +optional
```

Comment:
can we avoid duplicating logic of the `optional` fields, meaning drop the `+optional` marker having the `omitempty` tag? for most of the types this is redundant

also it would be great if other markers were placed as a top comment just to make the comments prettier (please check out the `Template` field in this struct as an example)

```
// +marker

// Comment
Field type `tags`
```

Applicable elsewhere

Reply:
I believe it should be good now